python/paddle/fluid/framework.py

```diff
@@ -4987,12 +4987,12 @@ def program_guard(main_program, startup_program=None):
              data = fluid.data(name='image', shape=[None, 784, 784], dtype='float32')
     
     """
-    if not isinstance(main_program, Program):
-        raise TypeError("main_program should be Program")
+    from .data_feeder import check_type
+    check_type(main_program, 'main_program', (Program), 'fluid.program_guard')
```

Comment:
Why parentheses "()" around Program ?

Reply:
`isinstance()` support single type or list of type. I keep same style with `()` of type. Here it's ok if remove `()`.

Reply:
But `(Program)` is just `Program`, `(Program, )` is a tuple contains only Program. I suggest to remove () or use `(Program, )`. Other things LGTM

Reply:
Thanks, fixed~~